homeassistant/components/select/device_condition.py

```diff
@@ -0,0 +1,88 @@
+"""Provide the device conditions for Select."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_CONDITION,
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_ENTITY_ID,
+    CONF_FOR,
+    CONF_TYPE,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import condition, config_validation as cv, entity_registry
+from homeassistant.helpers.config_validation import DEVICE_CONDITION_BASE_SCHEMA
+from homeassistant.helpers.typing import ConfigType, TemplateVarsType
+
+from .const import ATTR_OPTIONS, CONF_OPTION, DOMAIN
+
+CONDITION_TYPES = {"selected_option"}
+
+CONDITION_SCHEMA = DEVICE_CONDITION_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): vol.In(CONDITION_TYPES),
+        vol.Required(CONF_OPTION): str,
+        vol.Optional(CONF_FOR): cv.positive_time_period_dict,
+    }
+)
+
+
+async def async_get_conditions(
+    hass: HomeAssistant, device_id: str
+) -> list[dict[str, str]]:
+    """List device conditions for Select devices."""
+    registry = await entity_registry.async_get_registry(hass)
+    return [
+        {
+            CONF_CONDITION: "device",
+            CONF_DEVICE_ID: device_id,
+            CONF_DOMAIN: DOMAIN,
+            CONF_ENTITY_ID: entry.entity_id,
+            CONF_TYPE: "selected_option",
+        }
+        for entry in entity_registry.async_entries_for_device(registry, device_id)
+        if entry.domain == DOMAIN
+    ]
+
+
+@callback
+def async_condition_from_config(
+    config: ConfigType, config_validation: bool
+) -> condition.ConditionCheckerType:
+    """Create a function to test a device condition."""
+    if config_validation:
+        config = CONDITION_SCHEMA(config)
+
+    @callback
+    def test_is_state(hass: HomeAssistant, variables: TemplateVarsType) -> bool:
+        """Test if an entity is a certain state."""
+        return condition.state(
+            hass, config[CONF_ENTITY_ID], config[CONF_OPTION], config.get(CONF_FOR)
+        )
+
+    return test_is_state
+
+
+async def async_get_condition_capabilities(
+    hass: HomeAssistant, config: ConfigType
+) -> dict[str, Any]:
+    """List condition capabilities."""
+    state = hass.states.get(config[CONF_ENTITY_ID])
+    if state is None:
+        return {}
```

Comment:
This will break when the select entity is not in the state machine.
Since this is a capability attribute, use helper `get_capability` instead of directly referring to the state.

Reply:
Fixed in #52047